### PR TITLE
boards: gd32f450v_start: add initial support

### PR DESCRIPTION
**Pull request by gmarull**
Add initial support for the GDF450V-START board. GPIO (LED/Button), PWM
and UART have been tested successfully.

### CODE REVIEW

File: boards/arm/gd32f450v_start/board.cmake [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) 2022, Teslabs Engineering S.L.
+# SPDX-License-Identifier: Apache-2.0
+
+board_runner_args(jlink "--device=GD32F450VK" "--speed=4000")
+
+include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
```
Comment (soburi) [inline boards/arm/gd32f450v_start/board.cmake:6]: It may be good to use pyOCD rather than OpenOCD.

Comment (gmarull) [reply boards/arm/gd32f450v_start/board.cmake:6]: in this case openocd works fine (using stm32 configs really), so it doesn't hurt adding support for it. I'll also add pyocd.

Comment (soburi) [reply boards/arm/gd32f450v_start/board.cmake:6]: I got it. Let's keep it.

**Comment by gmarull** [comment]
@soburi added pyocd support as well.

**Review by nandojve — APPROVED** [review]

File: boards/arm/gd32f450v_start/gd32f450v_start.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2022, Teslabs Engineering S.L.
+# SPDX-License-Identifier: Apache-2.0
+
+identifier: gd32f450v_start
+name: GigaDevice GD32F450V-START
+type: mcu
+arch: arm
+ram: 256
```
Comment (nandojve) [inline boards/arm/gd32f450v_start/gd32f450v_start.yaml:8]: I remembered that SRAM0 is only 112k.

**Review by nandojve — CHANGES_REQUESTED** [review]

File: boards/arm/gd32f450v_start/gd32f450v_start.yaml (hunk 2 above)
Comment (gmarull) [reply boards/arm/gd32f450v_start/gd32f450v_start.yaml]: fixed

**Review by nandojve — APPROVED** [review]

**Review by soburi — APPROVED** [review]

**Pull request merged**